cpp/include/cuvs/cluster/spectral_clustering.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
```

Comment:
I propose we name this file `spectral.hpp` instead of `spectral_clustering`. The clustering is already in the namespace, so it's redundant.